manifests/init.pp

```diff
@@ -241,57 +255,57 @@
-  String $ap_req_checksum_type      = '',
-  String $safe_checksum_type        = '',
-  String $preferred_preauth_types   = '',
-  String $ccache_type               = '',
-  $canonicalize                     = '',
-  $dns_canonicalize_hostname        = '',
-  $dns_lookup_kdc                   = '',
-  $dns_lookup_realm                 = '',
-  $dns_fallback                     = '',
-  String $realm_try_domains         = '',
-  $extra_addresses                  = [],
-  String $udp_preference_limit      = '',
-  $verify_ap_req_nofail             = '',
-  String $ticket_lifetime           = '',
-  String $renew_lifetime            = '',
-  $noaddresses                      = '',
-  $forwardable                      = '',
-  $proxiable                        = '',
-  $rdns                             = '',
-  $pkinit_anchors                   = '',
-  $spake_preauth_groups             = '',
-  String $plugin_base_dir           = '',
-  $include                          = '',
-  $includedir                       = '',
-  $module                           = '',
-  String $db_module_dir             = '',
-  String $krb5_conf_path            = '/etc/krb5.conf',
-  String $krb5_conf_owner           = 'root',
-  String $krb5_conf_group           = 'root',
-  String $krb5_conf_mode            = '0444',
-  Boolean $alter_etc_services       = false,
-  Boolean $krb5_conf_warn           = true,
-  Hash $domain_realms               = {},
-  Hash $capaths                     = {},
-  Hash $appdefaults                 = {},
-  Hash $realms                      = {},
-  Hash $dbmodules                   = {}
+class mit_krb5 (
+  Optional[String] $default_realm              = undef,
+  Optional[String] $default_keytab_name        = undef,
+  Array $default_tgs_enctypes                  = [],
+  Array $default_tkt_enctypes                  = [],
+  Optional[String] $default_ccache_name        = undef,
+  Array $permitted_enctypes                    = [],
+  Optional[Boolean] $allow_weak_crypto         = undef,
+  Optional[String] $clockskew                  = undef,
+  Optional[String] $ignore_acceptor_hostname   = undef,
+  Optional[String] $k5login_authoritative      = undef,
+  Optional[String] $k5login_directory          = undef,
+  Optional[String] $kdc_timesync               = undef,
+  Optional[String] $kdc_req_checksum_type      = undef,
+  Optional[String] $ap_req_checksum_type       = undef,
+  Optional[String] $safe_checksum_type         = undef,
+  Optional[String] $preferred_preauth_types    = undef,
+  Optional[String] $ccache_type                = undef,
+  Optional[String] $canonicalize               = undef,
+  Optional[Boolean] $dns_canonicalize_hostname = undef,
+  Optional[Boolean] $dns_lookup_kdc            = undef,
+  Optional[Boolean] $dns_lookup_realm          = undef,
+  Optional[Boolean] $dns_fallback              = undef,
+  Optional[String] $realm_try_domains          = undef,
+  Array $extra_addresses                       = [],
```

Comment:
Should be `Array[Stdlib::IP::Address]` .

Note to self: the current _join logic_ used for `extra_addresses` is using space. `krb5.conf(5)` states that commas must be used.